javaslang/src/main/java/javaslang/Value.java

```diff
@@ -758,7 +758,7 @@ default CharSeq toCharSeq() {
      */
     default <K extends Comparable<? super K>, V> SortedMap<K, V> toSortedMap(Function<? super T, ? extends Tuple2<? extends K, ? extends V>> f) {
         Objects.requireNonNull(f, "f is null");
-        return toSortedMap((Comparator<? super K> & Serializable) K::compareTo, f);
+        return toSortedMap(Comparator.naturalOrder(), f);
```

Comment:
This falls into the category of the above comment.

What if we remove `? super` in the cast?

```java
return toSortedMap((Comparator<K> & Serializable) K::compareTo, f);
```